File: multiplesOf3and5.js

var Set = require('./set')
var multiplesOf3 = new Set();
var multiplesOf5 = new Set();

for (i=1; i<=999; i++) {
  if (i%3 == 0) {
    multiplesOf3.add(i);
    //console.log(i);
  }
  if (i%5 == 0) {
    multiplesOf3.add(i);
    //console.log(i);
  }
}

var all = multiplesOf3.union(multiplesOf5);
var list = all.get();
var addEmUp = 0;

for (i=0; i<=(list.length-1); i++) {
  addEmUp += parseInt(list[i]);
  //console.log(list[i]);
}

console.log(addEmUp);

File: multiplesOf3and5.R
################################################################################
# Usage: shell:
#   RScript multiplesOf3and5.R
# Usage: REPL:
#   r
#   > source('multiplesOf3and5.R')
################################################################################

multiplesOf3 <- vector(mode="numeric", length=0)
multiplesOf5 <- vector(mode="numeric", length=0)

for (i in 1:999 ) {
  if ( i %% 3 == 0) {
    multiplesOf3 <- c(multiplesOf3, i)
    #print(i)
  }
  if ( i %% 5 == 0) {
    multiplesOf5 <- c(multiplesOf5, i)
    #print(i)
  }
}

all <- union(multiplesOf3, multiplesOf5)
addEmUp <- 0
for (v in all) {
  addEmUp <- addEmUp + v
  #print(v)
}

print(addEmUp)

